shop/src/main/java/ro/msg/learning/shop/domain/Product.java

```diff
@@ -0,0 +1,56 @@
+package ro.msg.learning.shop.domain;
+
+import jakarta.persistence.*;
+import lombok.*;
+import org.antlr.v4.runtime.misc.NotNull;
+import org.hibernate.annotations.UuidGenerator;
+
+import java.math.BigDecimal;
+import java.util.Set;
+import java.util.UUID;
+
+@Data
+@Entity
```

Comment:
The `@Data` annotation contains `@ToString, @EqualsAndHashCode, @Getter / @Setter and @RequiredArgsConstructor together`. So we need to assure that we use this annotation only when we want all sub-annotations that are contained.
Please check here and in all other files to only use the needed annotations from lombok.
Reference: https://projectlombok.org/features/Data